### PR TITLE
Add GitHub utils 1

### PR DESCRIPTION
**Pull request by UnknownExecute**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?

**Type of Changes:**

 
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**
-  There is currently no way to retrieve user and calendar data from GitHub. #6

**What is the new behavior (if this is a feature change)?**
- There is now a way to retrieve user and calendar data from GitHub. This is done in an object. #6

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -18,6 +18,8 @@ import {
 // Routes
 import Routes from './Routes';
 
+// Tests
+import './tests';
```
Comment (Aichnerc) [inline src/App.js]: Please describe what this imported file is for using a multi-line commit.

File: src/tests.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// Import
+import * as gitHub from "./utilities/GitHubUtils"
```
Comment (Aichnerc) [inline src/tests.js]: Please describe what this imported file is for.

File: src/utilities/GitHubUtils/gqlData.js [hunk 3]
```diff
@@ -0,0 +1,238 @@
+//> Connect to backend
+
+//Import
+import { gql } from "apollo-boost";
```
Comment (Aichnerc) [inline src/utilities/GitHubUtils/gqlData.js]: Please follow our commenting guidelines for apollo connections. (See index.js)

File: src/utilities/GitHubUtils/index.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+//import './connectgql';
+
+export function get(username) {
+    return connect.get(username)
+}
+
+//export const test = (username) => (connect.get(username));
+
+// ConnectivityTest start
+
```
Comment (Aichnerc) [inline src/utilities/GitHubUtils/index.js]: Please remove the white-space between the comment and the function.

File: src/utilities/GitHubUtils/index.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+// index.js File
+
+// imports
+import * as connect from "./connectgql"
```
Comment (Aichnerc) [inline src/utilities/GitHubUtils/index.js]: Please describe what this imported file is for.

File: src/App.js [hunk 6]
```diff
@@ -18,6 +18,8 @@ import {
 // Routes
 import Routes from './Routes';
 
+// This IMPORT calls the const fetchDatafromGitHub variable. Wherein the get(username) is called
```
Comment (schettn) [inline src/App.js]: Please improve:
- No caps lock at "import".
- This comment does not represent the functionality of this import.

File: src/tests.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+// This IMPORT allows us to access the export GitHubUtils functions
```
Comment (schettn) [inline src/tests.js]: Please improve:

- No caps lock at "import".
- Refine the comment.

File: src/utilities/GitHubUtils/apolloclient.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+const LINK:HttpLink = new HttpLink({
+  uri: APILink,
+  headers: {
+    authorization: `Bearer ${
+      process.env.REACT_APP_GITHUB_PERSONAL_ACCESS_TOKEN
+    }`,
+  },
+});
+
+// Apollo Client setup
```
Comment (schettn) [inline src/utilities/GitHubUtils/apolloclient.js]: Client should be lowercase.

File: src/utilities/GitHubUtils/connectgql.js [hunk 9]
```diff
@@ -0,0 +1,32 @@
+//> Connect to backend
+import * as apollo from "./apolloclient";
+import * as gqlData from "./gqlData";
+
+// Get Profile and Calendar
```
Comment (schettn) [inline src/utilities/GitHubUtils/connectgql.js]: "Profile" and "Calendar" should be lowercase.

File: src/utilities/GitHubUtils/connectgql.js [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+
+    const {data} = resProfile;
+    const createdAtDate = new Date(data.user.createdAt);
+
+    const resCalendar = await apollo.client.query({
+      query: gqlData.GET_CALENDAR(username, createdAtDate),
+    });
+
+
+    const obj = {};
```
Comment (schettn) [inline src/utilities/GitHubUtils/connectgql.js]: Rename the variable "obj" to something which illustrates the meaning of the object.

File: src/utilities/GitHubUtils/gqlData.js [hunk 11]
```diff
@@ -0,0 +1,238 @@
+//> Connect to backend
+
+// Allows us to access QGL from apollo-boost
```
Comment (schettn) [inline src/utilities/GitHubUtils/gqlData.js]: Please improve:

- Change "QGL" to "GQL"
- Don't use "us" in a comment.

File: src/utilities/GitHubUtils/gqlData.js [hunk 12]
```diff
@@ -0,0 +1,238 @@
...
+import { gql } from "apollo-boost";
+
+// Structure GQL Profile Code
```
Comment (schettn) [inline src/utilities/GitHubUtils/gqlData.js]: "Profile" and "Code" should be lowercase.

Comment (schettn) [reply src/utilities/GitHubUtils/gqlData.js]: This is the GQL profile query structure.

File: src/utilities/GitHubUtils/gqlData.js [hunk 13]
```diff
@@ -0,0 +1,238 @@
...
+                }
+            }
+            }
+        }
+    }
+  }
+}
+`;
+
+// Structure GQL Calendar Code
```
Comment (schettn) [inline src/utilities/GitHubUtils/gqlData.js]: "Calendar" an "Code" should be lowercase.

Comment (schettn) [reply src/utilities/GitHubUtils/gqlData.js]: This is the GQL calendar query.

File: src/utilities/GitHubUtils/gqlData.js [hunk 14]
```diff
@@ -0,0 +1,238 @@
...
+            membersWithRole(first: 100) {
+                totalCount
+                nodes {
+                name
+                login
+                avatarUrl
+                url
+                projectsUrl
+                }
+            }
```
Comment (schettn) [inline src/utilities/GitHubUtils/gqlData.js]: Please fix the indentations.

File: src/utilities/GitHubUtils/gqlData.js [hunk 15]
```diff
@@ -0,0 +1,238 @@
...
+      query += GET_CALENDAR_QUERY_PART(date.toJSON(),count);
+
+      date.setFullYear(date.getFullYear()-1);
+      count++;
+  }
+  return query;
+
+};
+
+// Get Calendar skeletal structure
```
Comment (schettn) [inline src/utilities/GitHubUtils/gqlData.js]: What does that mean?

Comment (schettn) [reply src/utilities/GitHubUtils/gqlData.js]: "Calendar" should be lowercase.

File: src/utilities/GitHubUtils/index.js [hunk 16]
```diff
@@ -0,0 +1,17 @@
+// index.js File
```
Comment (schettn) [inline src/utilities/GitHubUtils/index.js]: "File" should be lowercase.

File: src/tests.js (hunk 7 above)
Comment (schettn) [reply src/tests.js]: Don't use "us".

File: src/utilities/GitHubUtils/index.js [hunk 17]
```diff
@@ -0,0 +1,17 @@
...
+
+// Enables us to access the export functions of connectqgl.js
```
Comment (schettn) [inline src/utilities/GitHubUtils/index.js]: Please fix:

- Don't use "us" in a comment.
- Refine the meaning of the comment.

File: src/App.js [hunk 18]
```diff
@@ -18,7 +18,7 @@ import {
 // Routes
 import Routes from './Routes';
 
-// This IMPORT calls the const fetchDatafromGitHub variable. Wherein the get(username) is called
+// This import loads tests.js into App.js
```
Comment (Aichnerc) [inline src/App.js]: I would actually want a comment stating why this is necessary and what it is used for.

File: src/tests.js [hunk 19]
```diff
@@ -1,4 +1,4 @@
-// This IMPORT allows us to access the export GitHubUtils functions
+// This import loads GItHubUtils in gitHub
```
Comment (Aichnerc) [inline src/tests.js]: I would actually want a comment stating why this is necessary and what it is used for.

File: src/utilities/GitHubUtils/index.js [hunk 20]
```diff
@@ -1,6 +1,6 @@
-// index.js File
+// index.js file
```
Comment (Aichnerc) [inline src/utilities/GitHubUtils/index.js]: What is this comment used for? There is no need to write the document name in the document.

File: src/App.js [hunk 21]
```diff
@@ -18,6 +18,9 @@ import {
 // Routes
```
Comment (kleberbaum) [inline src/App.js]: Looks more like a headline. Please use //> instead.

File: src/utilities/GitHubUtils/apolloclient.js [hunk 22]
```diff
@@ -0,0 +1,43 @@
+//> Connect to backend
+// Apollo
```
Comment (kleberbaum) [inline src/utilities/GitHubUtils/apolloclient.js]: Looks more like a headline. Please use //> instead.

File: src/utilities/GitHubUtils/apolloclient.js [hunk 23]
```diff
@@ -0,0 +1,43 @@
+//> Connect to backend
```
Comment (kleberbaum) [inline src/utilities/GitHubUtils/apolloclient.js]: Looks more like a comment. Unnecessary headline please remove.

File: src/utilities/GitHubUtils/connectgql.js [hunk 24]
```diff
@@ -0,0 +1,37 @@
...
+    });
+
+    const {data} = resProfile;
+    const createdAtDate = new Date(data.user.createdAt);
+
+    const resCalendar = await apollo.client.query({
+      query: gqlData.getCalendar(username, createdAtDate),
+    });
+
+
```
Comment (kleberbaum) [inline src/utilities/GitHubUtils/connectgql.js]: Please remove unnecessary newline.

File: src/utilities/GitHubUtils/apolloclient.js [hunk 25]
```diff
@@ -0,0 +1,43 @@
...
+    __schema: {
+      types: [], // no types provided - works like a charm.ing
+    },
+  },
+});
+const cache = new InMemoryCache({ fragmentMatcher });
+
+// Create api url from base url
+const APILink = APIHost+"/graphql";
+//const APILink = APIHost;
```
Comment (kleberbaum) [inline src/utilities/GitHubUtils/apolloclient.js]: Unnecessary please remove.

File: src/utilities/GitHubUtils/gqlData.js [hunk 26]
```diff
@@ -0,0 +1,243 @@
+//> Connect to backend
```
Comment (kleberbaum) [inline src/utilities/GitHubUtils/gqlData.js]: Looks more like a comment. Unnecessary headline please remove.

File: src/utilities/GitHubUtils/gqlData.js [hunk 27]
```diff
@@ -0,0 +1,243 @@
...
+// Get calendar basic structure
+export const getCalendar = (username, createdAt) => {
+  const query = gql`
+                  query
+                  {
+                      user(login: "${username}") {
+                          ${generateCalendarsQuery(username, createdAt)}
+                      }
+                  }
+  `;
```
Comment (kleberbaum) [inline src/utilities/GitHubUtils/gqlData.js]: Indentation doesn't seem right.

File: src/utilities/GitHubUtils/gqlData.js [hunk 28]
```diff
@@ -0,0 +1,243 @@
...
+              }
+            }
+          }
+        }
+      }
+    }
+  `;
+};
+
+// Dynamic generate calendars structure
```
Comment (kleberbaum) [inline src/utilities/GitHubUtils/gqlData.js]: Comment doesn't seem quite right.

File: src/utilities/GitHubUtils/apolloclient.js (hunk 22 above)
Comment (Aichnerc) [reply src/utilities/GitHubUtils/apolloclient.js]: Nah, "Connection to backend" is the title, Apollo is the subheading.

File: src/utilities/GitHubUtils/gqlData.js (hunk 26 above)
Comment (Aichnerc) [reply src/utilities/GitHubUtils/gqlData.js]: The entire comment is unnecessary.

File: src/utilities/GitHubUtils/apolloclient.js (hunk 23 above)
Comment (kleberbaum) [reply src/utilities/GitHubUtils/apolloclient.js]: At the moment we do not have a backend (only the github api) and even if we had one, that would not be enough as a title.

File: src/utilities/GitHubUtils/apolloclient.js (hunk 22 above)
Comment (kleberbaum) [reply src/utilities/GitHubUtils/apolloclient.js]: At the moment we do not have a backend (only the github api) and even if we had one, that would not be enough as a title.

File: src/utilities/GitHubUtils/gqlData.js (hunk 26 above)
Comment (kleberbaum) [reply src/utilities/GitHubUtils/gqlData.js]: At the moment we do not have a backend (only the github api) and even if we had one, that would not be enough as a title.

File: src/utilities/GitHubUtils/apolloclient.js (hunk 22 above)
Comment (kleberbaum) [reply src/utilities/GitHubUtils/apolloclient.js]: At the moment we do not have a backend (only the github api) and even if we had one, that would not qualify as a title.

Comment (kleberbaum) [reply src/utilities/GitHubUtils/apolloclient.js]: At the moment we do not have a backend (only the github api) and even if we had one, that would not qualify as a good title.

File: src/utilities/GitHubUtils/apolloclient.js (hunk 23 above)
Comment (kleberbaum) [reply src/utilities/GitHubUtils/apolloclient.js]: At the moment we do not have a backend (only the github api) and even if we had one, that would not qualify as a good title.

File: src/utilities/GitHubUtils/gqlData.js (hunk 26 above)
Comment (kleberbaum) [reply src/utilities/GitHubUtils/gqlData.js]: At the moment we do not have a backend (only the github api) and even if we had one, that would not qualify as a good title.

File: src/App.test.js [hunk 29]
```diff
@@ -0,0 +1,19 @@
+// A tool to query data from github
```
Comment (schettn) [inline src/App.test.js]: Please change "github" to "GitHub".

File: src/utilities/GitHubUtils/index.js [hunk 30]
```diff
@@ -0,0 +1,11 @@
+// Builds a object from github user
```
Comment (schettn) [inline src/utilities/GitHubUtils/index.js]: Please change "github" to "GitHub"

**Pull request merged**